src/index.ts

```diff
@@ -120,9 +145,61 @@ const GlobalIdExtensionsTweakFieldsPlugin: Plugin = function(builder) {
       });
     }, fields);
   });
+
+  // add deprecations
+  builder.hook("GraphQLObjectType:fields:field", (field, { inflection }, context) => {
+    const {
+      scope: { pgFieldIntrospection }
+    } = context;
+
+    if (
+      !pgFieldIntrospection ||
+      !pgFieldIntrospection.class ||
+      pgFieldIntrospection.kind !== "attribute"
+    ) {
+      return field;
+    }
+
+    const attr: PgAttribute = pgFieldIntrospection;
+    const table: PgClass = pgFieldIntrospection.class;
+
+    if (containsSingleColumn(table.primaryKeyConstraint, attr)) {
+      return maybeDeprecate(field, attr, 'nodeId');
+    }
+
+    const fk = table.constraints.find(c =>
+      isForeignKey(c) &&
+      containsSingleColumn(c, attr)
+    );
+
+    if (fk) {
+      const fieldName = inflection.singleRelationByKeys(
+        fk.keyAttributes,
+        (fk as any).foreignClass,
```

Comment:
```suggestion
        fk.foreignClass,
```

Reply:
`Property 'foreignClass' does not exist on type 'PgConstraint'. Did you mean 'foreignClassId'?`

Reply:
Your version of PostGraphile might be out of date:

https://github.com/graphile/graphile-engine/blob/ec3bcc496c00fd4b25293c06d2a890ffc234cf8b/packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.d.ts#L118